File: 2022/day01/javascript/part1.mjs
import { join } from "node:path";
import { readFile } from "node:fs/promises";

const filePath = new URL(join(import.meta.url, "..", "..", "input.txt"));

const file = await readFile(filePath, { encoding: "utf-8" });
const data = file.split("\n");

const cal = [];
let current = 0;

for (let line of data) {
  if (line !== "") {
    current += parseInt(line);
  } else {
    cal.push(current);
    current = 0;
  }
}

console.log(Math.max(...cal));

File: 2022/day01/part1.py
with open('./input.txt') as f:
    data = f.read().splitlines()

    cal = []
    current = 0
    for i in data:
        if i != '':
            current += int(i)
        else:
            cal.append(current)
            current = 0

    highest = max(cal)
    print(highest)
